packages/rich-text/src/component/index.native.js

```diff
@@ -693,23 +704,18 @@ export class RichText extends Component {
 					console.warn( 'Oops, selection will land outside the text, skipping setting it...' );
 					selection = null;
 				} else {
-					// The following regular expression is used in Aztec here:
-					// https://github.com/wordpress-mobile/AztecEditor-Android/blob/b1fad439d56fa6d4aa0b78526fef355c59d00dd3/aztec/src/main/kotlin/org/wordpress/aztec/AztecParser.kt#L656
-					const brBeforeParaMatches = html.match( /(<br>)+<\/p>$/g );
-					if ( brBeforeParaMatches ) {
+					const count = this.countBreaksBeforeParagraphTag( html );
+					if ( count > 0 ) {
 						console.warn( 'Oops, BR tag(s) at the end of content. Aztec will remove them, adapting the selection...' );
-						const count = ( brBeforeParaMatches[ 0 ].match( /br/g ) || [] ).length;
-						if ( count > 0 ) {
-							let newSelectionStart = this.props.selectionStart - count;
-							if ( newSelectionStart < 0 ) {
-								newSelectionStart = 0;
-							}
-							let newSelectionEnd = this.props.selectionEnd - count;
-							if ( newSelectionEnd < 0 ) {
-								newSelectionEnd = 0;
-							}
-							selection = { start: newSelectionStart, end: newSelectionEnd };
+						let newSelectionStart = this.props.selectionStart - count;
+						if ( newSelectionStart < 0 ) {
+							newSelectionStart = 0;
+						}
+						let newSelectionEnd = this.props.selectionEnd - count;
+						if ( newSelectionEnd < 0 ) {
+							newSelectionEnd = 0;
 						}
+						selection = { start: newSelectionStart, end: newSelectionEnd };
```

Comment:
I really like how you have the regex broken out into a separate function with tests (those tests you added are 😻). 

I would have probably taken this a bit further and extracted out all or most of 707-718 into some kind of pure 'calculateSelectionBasedOnAztecModifications( html, { originalStartSelection, originalEndSelection } )` function that took in the html and old selection and output the updated selection for the anticipated Aztec-output text. We would then end up with just having something like this in the `render` method:

```
...
} else {
    selection = calculateSelectionBasedOnAztecModifications( html, selection );
}
```

With that said, I think extracting that out at this point falls well within the realm of personal preference, and the changes you have made are already a really great improvement, so feel free to leave this as-is if you prefer. 😄